### PR TITLE
fix message hasher and tests

### PR DESCRIPTION
**Pull request by makramkd**
## Motivation

Fix the message hasher EVM implementation after the chainlink-common and chainlink-ccip bumps which alter the `Message` type.

### CODE REVIEW

**Review by 0xnogo — APPROVED** [review]

File: core/services/ocr3/plugins/ccipevm/msghasher.go [hunk 1]
```diff
@@ -53,79 +50,58 @@ func NewMessageHasherV1(
...
-	if len(msg.TokenAmounts) != len(msg.SourceTokenData) {
-		return [32]byte{}, fmt.Errorf("token amounts and source token data must have the same length")
+func (h *MessageHasherV1) Hash(_ context.Context, msg cciptypes.Message) (cciptypes.Bytes32, error) {
+	var rampTokenAmounts []message_hasher.InternalRampTokenAmount
+	for _, rta := range msg.TokenAmounts {
+		rampTokenAmounts = append(rampTokenAmounts, message_hasher.InternalRampTokenAmount{
+			SourcePoolAddress: rta.SourcePoolAddress,
+			DestTokenAddress:  rta.DestTokenAddress,
+			ExtraData:         rta.ExtraData,
+			Amount:            rta.Amount.Int,
```
Comment (elatoskinas) [inline core/services/ocr3/plugins/ccipevm/msghasher.go:63]: q: this is a big int right?

File: core/services/ocr3/plugins/ccipevm/msghasher.go [hunk 2]
```diff
@@ -149,5 +125,27 @@ func (h *MessageHasherV1) abiEncode(method string, values ...interface{}) ([]byt
 	return res[4:], nil
 }
 
+func decodeExtraArgs(extraArgs []byte) (gasLimit *big.Int, err error) {
+	var method string
+	if bytes.Equal(extraArgs[:4], evmExtraArgsV1Tag) {
```
Comment (elatoskinas) [inline core/services/ocr3/plugins/ccipevm/msghasher.go:130]: suggestion: once we support more chains, might be easier to just create a map `bytes -> string` and just lookup the value. We could maintain that constants file in the same file as we would do chain selector family lookups

File: core/services/ocr3/plugins/ccipevm/msghasher.go (hunk 1 above)
Comment (makramkd) [reply core/services/ocr3/plugins/ccipevm/msghasher.go:63]: Yep, its a `*big.Int` which is what `uint256` is translated to in Go.

File: core/services/ocr3/plugins/ccipevm/msghasher.go (hunk 2 above)
Comment (makramkd) [reply core/services/ocr3/plugins/ccipevm/msghasher.go:130]: Yeah, thats a good point. I think there's quite a bit of room for improvement in terms of code organization (and possibly these interfaces) after we get the basic functionality working.

Comment (makramkd) [reply core/services/ocr3/plugins/ccipevm/msghasher.go:130]: The order of operations for parsing the extra args would be something like:

* Look up the chain family for the source chain and dest chain selectors
* Decode the extra args with the chain-family specific extra args decoder, return the decoded values for the dest chain
  * Within each extra args decoder, you'd have some kind of table, for EVM we'd have a table of valid tags, for others we'd probably want to follow a similar pattern.

**Pull request merged**